### PR TITLE
Csp nonce - Security! - Ready for Review/Merge

### PR DESCRIPTION
**Pull request by JohnMcLear**
This provides CSP functionality inside of Etherpad

Etherpad has a lot of inline JS, instead of removing it (which would be nice but not required) CSP provides the functionality to generate a nonce (generated using the ep_helmet plugin) and providing that to the page.

Pages tested and fixed:
- [x] Index
- [x] Pad
- [x] Timeslider
- [x] Pad Import
- [x] Admin pages
- [x] Test pages (known broken but WIP on another branch so don't want to work on it)
- [x] A bunch of plugins.
- [ ] Ace has a placeholder Nonce in for when it renders the outerHTML to the page.  I can't find a good way for this but I'm not actually convinced it's exploitable...

Required for https://github.com/ether/etherpad-lite/issues/3094

### CODE REVIEW

**Comment by JohnMcLear** [comment]
To get the actual nonce values ep_helmet is required, else no nonce is provided.

File: src/static/js/ace.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ var _ = require('./underscore');
 
 function scriptTag(source) {
   return (
-    '<script type="text/javascript">\n'
+    '<script type="text/javascript" nonce="aceTemporaryNonce">\n'
```
Comment (rhansen) [inline src/static/js/ace.js:38]: Shouldn't this nonce come from `settings.nonce`?

File: src/templates/index.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
 <html>
 
         <title><%=settings.title%></title>
-    <script>
+    <script nonce="<%=settings.nonce && settings.nonce.padLicense || ''%>">
```
Comment (rhansen) [inline src/templates/index.html:8]: If `settings.nonce.padLicense` is nullish/empty, wouldn't it be better to not add the `nonce` attribute? Like this:

```suggestion
    <script <%=(settings.nonce && settings.nonce.padLicense) ? `nonce="${settings.nonce.padLicense}"` : ''%>>
```

Comment (rhansen) [inline src/templates/index.html:8]: A nonce should only be used once: refreshing should get a different nonce, and two different users should never see the same nonce. If the same nonce is used every time, an attacker could visit the page, look at the nonce, then use it in the injected code.

**Comment by JohnMcLear** [comment]
Closing as stale.

**Pull request closed without merging**